### PR TITLE
Add helm runtime provider for kubernetes

### PR DESCRIPTION
**Pull request by Hevienz**

### CODE REVIEW

File: pkg/manager/job/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
+package job
```
Comment (chilianyi) [inline pkg/manager/job/kubernetes.go]: Could this file be moved to plugin?

File: pkg/plugins/helm/parser.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+		return nil, err
+	}
+
+	cluster, err := p.ParseCluster(vals, versionId)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterWrapper := &models.ClusterWrapper{
+		Cluster: cluster,
```
Comment (chilianyi) [inline pkg/plugins/helm/parser.go]: ClusterNode,  ClusterRole, ClusterLinks may be needed.

File: pkg/manager/job/kubernetes.go (hunk 1 above)
Comment (Hevienz) [reply pkg/manager/job/kubernetes.go]: Ok, I will remove it to plugin.

File: pkg/plugins/helm/parser.go (hunk 2 above)
Comment (Hevienz) [reply pkg/plugins/helm/parser.go]: I have no idea about which concept in k8s can be mapping as ClusterRole?

Comment (chilianyi) [reply pkg/plugins/helm/parser.go]: A deployment is a role, a pod is a node, 
cpu, memory, env(defined in values.yaml, each role has the same env) should be stored into ClusterRole.

Comment (Hevienz) [reply pkg/plugins/helm/parser.go]: Ok

File: pkg/plugins/helm/provider.go [hunk 3]
```diff
@@ -29,11 +41,35 @@ func init() {
 type Provider struct {
 }
 
-func (p *Provider) getKubeClient() (clientset *kubernetes.Clientset, config *rest.Config, err error) {
-	config, err = clientcmd.BuildConfigFromFlags("", homedir.HomeDir()+"/.kube/config")
+func (p *Provider) getKubeClient(runtimeId string) (clientset *kubernetes.Clientset, config *rest.Config, err error) {
+	kubeconfigGetter := func() (*clientcmdapi.Config, error) {
+		ctx := context.Background()
+		client, err := runtimeclient.NewRuntimeManagerClient(ctx)
```
Comment (chilianyi) [inline pkg/plugins/helm/provider.go]: Here can use runtimeclient.NewRuntime(runtimeId) instead.

File: pkg/plugins/helm/provider.go [hunk 4]
```diff
@@ -70,21 +106,367 @@ func (p *Provider) getHelmClient() (helmClient *helm.Client, err error) {
...
+		pkg := resp.GetPackage()
+		r := bytes.NewReader(pkg)
+
+		c, err := chartutil.LoadArchive(r)
+		if err != nil {
+			return err
+		}
+
+		_, err = hc.InstallReleaseFromChart(c, taskDirective.Namespace,
+			helm.ValueOverrides([]byte(c.Values.GetRaw())),
```
Comment (chilianyi) [inline pkg/plugins/helm/provider.go]: If values is changed by user when create cluster, here need to set new values into c.Values

File: pkg/plugins/helm/parser.go [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+		return nil, err
+	}
+
+	clusterLinks, err := p.ParseClusterLinks(c, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterWrapper := &models.ClusterWrapper{
+		Cluster:      cluster,
```
[the remark this answers is not in the document]
Comment (Hevienz) [reply pkg/plugins/helm/parser.go]: ClusterNodes need to be updated when helm release is updated.
So I add UpdateClusterStatus for runtime provider interface and call it when job is done.

File: pkg/plugins/helm/provider.go (hunk 4 above)
Comment (Hevienz) [reply pkg/plugins/helm/provider.go]: I think I need a field on Cluster to store the values.

File: pkg/plugins/helm/provider.go [hunk 6]
```diff
@@ -53,37 +80,424 @@ func (p *Provider) setupHelm(kubeClient *kubernetes.Clientset, restClientConfig
...
+	}
+
+	if len(resp.RuntimeSet) == 0 {
+		return fmt.Errorf("Runtime [%s] not found", runtimeId)
+	}
+
+	runtime := resp.RuntimeSet[0]
+	namespace := runtime.GetZone().GetValue()
+
+	pbClusterNodes, err := p.getKubePodsAsClusterNodes(runtimeId, namespace, job.ClusterId, job.Owner)
```
Comment (chilianyi) [inline pkg/plugins/helm/provider.go]: Modify use update sql, can not insert new rows. 
We may need a "INSERT ON DUPLICATE KEY UPDATE" func in pkg/db. 
@zheng1 Is there such func in dbr?

File: pkg/plugins/helm/directive.go [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+func getJobDirective(data string) (*JobDirective, error) {
+	clusterWrapper, err := models.NewClusterWrapper(data)
+	if err != nil {
+		return nil, err
+	}
+
+	runtimeId := clusterWrapper.Cluster.RuntimeId
+
+	ctx := context.Background()
+	client, err := runtimeclient.NewRuntimeManagerClient(ctx)
```
Comment (chilianyi) [inline pkg/plugins/helm/directive.go]: here

**Pull request merged**